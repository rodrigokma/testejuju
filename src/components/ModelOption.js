function ModelOption(props) {
  return (
    <div className="model-option">
      {props.brandCode && (
        <select value={props.value} onChange={props.onChange}>
          <option data-testid="select-model" value="">
            Choose a model
          </option>

          {props.models.map((model) => {
            return (
              <option key={`model-${model.codigo}`} value={model.codigo}>
                {model.nome}
              </option>
            );
          })}
        </select>
      )}
    </div>
  );
}

export default ModelOption;
