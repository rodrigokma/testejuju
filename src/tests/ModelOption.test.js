import React from "react";
import ModelOption from "../components/ModelOption";
import { render } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import TestRenderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

describe("Model options component", () => {
  const modelOptionJsx = (
    <ModelOption
      brandCode="93"
      onChange={jest.fn()}
      models={[
        { nome: "corsa", codigo: "72" },
        { nome: "celta", codigo: "43" },
        { nome: "malibu", codigo: "32" },
      ]}
    />
  );

  const component = shallow(modelOptionJsx);

  test("renders component", () => {
    expect(component.exists()).toBe(true);
  });

  test("matches the snapshot", () => {
    const componentSnapshot = TestRenderer.create(modelOptionJsx);
    expect(componentSnapshot).toMatchSnapshot();
  });

  test("has a 'Select a model' as first option", () => {
    const { getByTestId } = render(modelOptionJsx);

    expect(getByTestId("select-model")).toHaveTextContent("Choose a model");
  });

  test("renders three model options", () => {
    expect(component.find("option")).toHaveLength(4);
  });
});
