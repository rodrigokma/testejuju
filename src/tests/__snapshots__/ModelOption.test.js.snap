// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Model options component matches the snapshot 1`] = `
<div
  className="model-option"
>
  <select
    onChange={[MockFunction]}
  >
    <option
      data-testid="select-model"
      value=""
    >
      Choose a model
    </option>
    <option
      value="72"
    >
      corsa
    </option>
    <option
      value="43"
    >
      celta
    </option>
    <option
      value="32"
    >
      malibu
    </option>
  </select>
</div>
`;
